src/main/java/core/basesyntax/service/impl/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        Integer balance = Storage.storage.getOrDefault(transaction.getFruit(), 0);
+        if (balance != 0) {
+            Integer result = balance - transaction.getQuantity();
+            Storage.storage.put(transaction.getFruit(), result);
+        } else {
+            throw new RuntimeException("Their is no " + transaction.getFruit() + " in fruit shop");
+        }
```

Comment:
What if Storage balance == 5 and transaction.getQuantity() == 25. Is it ok?
![image](https://user-images.githubusercontent.com/20326858/227502592-ee9e8677-0a1a-4684-a366-97c33a479829.png)